### ISSUE TITLE
[internal] Broken test on diff: `error: no changes were expected but changes were proposed`

### ISSUE THREAD

**Issue by guineveresaenger** [pulumi/pulumi-artifactory#442]
[Link to recent Action run](https://github.com/pulumi/pulumi-artifactory/actions/runs/6127399453/job/16633182212#step:22:180)

This is blocking upgrades.

I suspect this may be a diff bug in the bridge since it started happening on [this upgrade attempt](https://github.com/pulumi/pulumi-artifactory/pull/436). Linking stack output after a `pulumi up` in a comment  to avoid cluttering this description; note that `projectKey`, the Input, is set to `default` but the Output is `""`.

**Comment by guineveresaenger** [comment]
`stack.json` after `pulumi up`:

```
{
    "version": 3,
    "deployment": {
        "manifest": {
            "time": "2023-09-07T17:27:38.552557-07:00",
            "magic": "f7e78335a73b6584d4a30dcb74dbbcb4a83203946d15dc55fe9360f1e457dcff",
            "version": "v3.81.0"
        },
        "secrets_providers": {
            "type": "service",
            "state": {
                "url": "https://api.pulumi.com",
                "owner": "guinevere-pulumi-corp",
                "project": "artifactory-tests",
                "stack": "dev"
            }
        },
        "resources": [
            {
                "urn": "urn:pulumi:dev::artifactory-tests::pulumi:pulumi:Stack::artifactory-tests-dev",
                "custom": false,
                "type": "pulumi:pulumi:Stack",
                "created": "2023-09-08T00:27:36.94448Z",
                "modified": "2023-09-08T00:27:36.94448Z",
                "sourcePosition": "project:///node_modules/@pulumi/runtime/stack.ts#36,23"
            },
            {
                "urn": "urn:pulumi:dev::artifactory-tests::pulumi:providers:artifactory::default_4_6_0_alpha_1694121457_ec7ab1c7_dirty",
                "custom": true,
                "id": "260d762e-6d7b-4c4c-b92e-e166f8bba017",
                "type": "pulumi:providers:artifactory",
                "inputs": {
                    "accessToken": {
                        "4dabf18193072939515e22adb298388d": "1b47061264138c4ac30d75fd1eb44270",
                        "ciphertext": "AAABAMQJNcs79aBGk81NoVIMN2wT6uV8gudv8YlP9DYLzhuid7UmxHFBD29CWlogfrKryL+KT6bENWZ6WqrBLhdHJVfeYQ+8L2hRY2jYBVDBmlF13cU7sqjiPYjNAo2k7tgK00zjC5htKnnuFFAVUIspFBXd0VoZl3Rsm85w5hJRB7EAGSdLje7K7jVv/gNw1vLG7jV8+3CleKdgXc+s+BTVQI/8toxmBdNLZXd1Ii5jeCmPpdCUOnIrSeW4w6lIIBC7GBpOwjHk/4QDE/dMeyDQGWYg/bfk/22UYkU4qrPXoa8kzzvGvl/hFCmLsKThawER+NVlKnT0tfMGbqwL7KtFEjyGZb0UVsjK65Kd+a4AXZ0vH2IKhEYz0yzi04hD7CBU5mQ7szRiHkEBAKB10nPbvQQUrtDbo460mbiupP/9J7YOYXiFhZ1Mn6ijwL4tWzPeFC+kLjVYUM7ihuIJDIgo53INpljsvQmqqHUdX3eRij7xiMy//f+GJD8eZM3bCJKkFpo8f/DKiCplcxVLLouB8D7qY1oMamd9MOPJgu00WT+NI4lPRGR3iMs4Nn0qJXvdahs+DsCsJjrHOTxpmU4XyvYzmRQuEfu9ds6RXAY2TBv42O60+Rfc+G5x0xNfs+ofGnv63r4olswYYJywdLiqJAtjPpiXmY28JvIcsX9VvdK1J7e1Lw02ufV0dMn3CmS+isSIDT3WoF0JDhj35kck6NGtrkDEMF24rp/2Lm46ARRl/2Oo1vRCGb56OgnfHFIie5iBwSQ8lG6P+h/I1noNvb5gq9rNfGA/WgEZqA1bDzqK2kzBMi1WCxDtelhYMoDMWtULZER+iLZIIbVQdEd80IeQzK4FQ5WJWKgaRiGzl9JdkelQIdQO4tSW/dGlBvXdKw1jxZxl4XtrmKik4TJb2T4IzsFB4CVel44ok4AmZ+ILxgMvTZDrUVtbADv0zh8f7ceoRvqMMtw7q3GkTk9FmwctV2vrpBQuv9fljOXE+v9jFUXJlDYPB3Umqess"
                    },
                    "checkLicense": "false",
                    "url": "https://pulumitest.jfrog.io",
                    "version": "4.6.0-alpha.1694121457+ec7ab1c7.dirty"
                },
                "outputs": {
                    "accessToken": {
                        "4dabf18193072939515e22adb298388d": "1b47061264138c4ac30d75fd1eb44270",
                        "ciphertext": "AAABAMQJNcs79aBGk81NoVIMN2wT6uV8gudv8YlP9DYLzhuid7UmxHFBD29CWlogfrKryL+KT6bENWZ6WqrBLhdHJVfeYQ+8L2hRY2jYBVDBmlF13cU7sqjiPYjNAo2k7tgK00zjC5htKnnuFFAVUIspFBXd0VoZl3Rsm85w5hJRB7EAGSdLje7K7jVv/gNw1vLG7jV8+3CleKdgXc+s+BTVQI/8toxmBdNLZXd1Ii5jeCmPpdCUOnIrSeW4w6lIIBC7GBpOwjHk/4QDE/dMeyDQGWYg/bfk/22UYkU4qrPXoa8kzzvGvl/hFCmLsKThawER+NVlKnT0tfMGbqwL7KtFEjyGZb0UVsjK65Kd+a4AXZ0vH2IKhEYz0yzi04hD7CBU5mQ7szRiHkEBAKB10nPbvQQUrtDbo460mbiupP/9J7YOYXiFhZ1Mn6ijwL4tWzPeFC+kLjVYUM7ihuIJDIgo53INpljsvQmqqHUdX3eRij7xiMy//f+GJD8eZM3bCJKkFpo8f/DKiCplcxVLLouB8D7qY1oMamd9MOPJgu00WT+NI4lPRGR3iMs4Nn0qJXvdahs+DsCsJjrHOTxpmU4XyvYzmRQuEfu9ds6RXAY2TBv42O60+Rfc+G5x0xNfs+ofGnv63r4olswYYJywdLiqJAtjPpiXmY28JvIcsX9VvdK1J7e1Lw02ufV0dMn3CmS+isSIDT3WoF0JDhj35kck6NGtrkDEMF24rp/2Lm46ARRl/2Oo1vRCGb56OgnfHFIie5iBwSQ8lG6P+h/I1noNvb5gq9rNfGA/WgEZqA1bDzqK2kzBMi1WCxDtelhYMoDMWtULZER+iLZIIbVQdEd80IeQzK4FQ5WJWKgaRiGzl9JdkelQIdQO4tSW/dGlBvXdKw1jxZxl4XtrmKik4TJb2T4IzsFB4CVel44ok4AmZ+ILxgMvTZDrUVtbADv0zh8f7ceoRvqMMtw7q3GkTk9FmwctV2vrpBQuv9fljOXE+v9jFUXJlDYPB3Umqess"
                    },
                    "checkLicense": "false",
                    "url": "https://pulumitest.jfrog.io",
                    "version": "4.6.0-alpha.1694121457+ec7ab1c7.dirty"
                },
                "created": "2023-09-08T00:27:38.342067Z",
                "modified": "2023-09-08T00:27:38.342067Z"
            },
            {
                "urn": "urn:pulumi:dev::artifactory-tests::artifactory:index/localGenericRepository:LocalGenericRepository::pulumipus-ts",
                "custom": true,
                "id": "default-pulumipus-ts",
                "type": "artifactory:index/localGenericRepository:LocalGenericRepository",
                "inputs": {
                    "__defaults": [
                        "blackedOut",
                        "cdnRedirect",
                        "priorityResolution",
                        "repoLayoutRef",
                        "xrayIndex"
                    ],
                    "blackedOut": false,
                    "cdnRedirect": false,
                    "key": "default-pulumipus-ts",
                    "priorityResolution": false,
                    "projectKey": "default",
                    "repoLayoutRef": "simple-default",
                    "xrayIndex": false
                },
                "outputs": {
                    "archiveBrowsingEnabled": false,
                    "blackedOut": false,
                    "cdnRedirect": false,
                    "description": "",
                    "downloadDirect": false,
                    "excludesPattern": "",
                    "id": "default-pulumipus-ts",
                    "includesPattern": "**/*",
                    "key": "default-pulumipus-ts",
                    "notes": "",
                    "packageType": "generic",
                    "priorityResolution": false,
                    "projectEnvironments": [],
                    "projectKey": "",
                    "propertySets": [],
                    "repoLayoutRef": "simple-default",
                    "xrayIndex": false
                },
                "parent": "urn:pulumi:dev::artifactory-tests::pulumi:pulumi:Stack::artifactory-tests-dev",
                "provider": "urn:pulumi:dev::artifactory-tests::pulumi:providers:artifactory::default_4_6_0_alpha_1694121457_ec7ab1c7_dirty::260d762e-6d7b-4c4c-b92e-e166f8bba017",
                "propertyDependencies": {
                    "key": null,
                    "projectKey": null
                },
                "sourcePosition": "project:///index.ts#4,1"
            }
        ]
    }
}
```

**Comment by mikhailshilkov** [comment]
I've opened two separate PRs for provider upgrade and bridge upgrade to bisect the root cause:
- https://github.com/pulumi/pulumi-artifactory/pull/443
- https://github.com/pulumi/pulumi-artifactory/pull/444

**Comment by guineveresaenger** [comment]
#441 should have tests now that pass.

TL; DR: the `projectKey` field always triggers an update when set to `default`. Propose to use `User` resource for tests.

**Comment by mikhailshilkov** [comment]
Fixed by https://github.com/pulumi/pulumi-artifactory/pull/441
